src/averageAges.js

```diff
@@ -15,11 +15,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const averageMan = people.filter(person =>
+    person.sex === 'm'
+    && (!century || Math.ceil(person.died / 100) === century));
+
+  const averageAges = averageMan.map(person => {
+    return person.died - person.born;
+  });
+
+  return averageAges.reduce((a, b) => a + b, 0) / averageAges.length;
```

Comment:
move to the separate functions & reuse in all cases
```suggestion
  const averageAges = getAverageAges(men)

  return averageAges;
```